core/valor_core/classification.py

```diff
@@ -0,0 +1,1343 @@
+                label_key=key, value=dict(value), pr_curve_iou_threshold=None
+            )
+            for key, value in detailed_pr_output.items()
+        ]
+
+    return output
+
+
+def _compute_clf_metrics(
+    groundtruth_df: pd.DataFrame,
+    prediction_df: pd.DataFrame,
+    unique_labels: list,
+    label_map: Optional[Dict[schemas.Label, schemas.Label]] = None,
+    metrics_to_return: Optional[List[enums.MetricType]] = None,
+    pr_curve_max_examples: int = 1,
+) -> Tuple[List[dict], List[dict]]:
+    """
+    Compute classification metrics including confusion matrices and various performance metrics.
+
+    Parameters
+    ----------
+    groundtruth_df : pd.DataFrame
+        DataFrame containing ground truth annotations with necessary columns.
+    prediction_df : pd.DataFrame
+        DataFrame containing predictions with necessary columns.
+    unique_labels : list
+        List of unique labels used for the classification.
+    label_map : Optional[Dict[schemas.Label, schemas.Label]], default=None
+        Optional dictionary mapping ground truth labels to prediction labels.
+    metrics_to_return : Optional[List[enums.MetricType]], default=None
+        List of metric types to return. If None, default metrics are used.
+    pr_curve_max_examples : int, default=1
+        Maximum number of examples to use for Precision-Recall curve calculations.
+
+    Returns
+    -------
+    Tuple[List[dict], List[dict]]
+        A tuple where:
+        - The first element is a list of dictionaries representing confusion matrices.
+        - The second element is a list of dictionaries representing the requested classification metrics.
+    """
+    grouper_mappings = _create_classification_grouper_mappings(
+        label_map=label_map,
+        labels=unique_labels,
+    )
+
+    confusion_matrices, metrics_to_output = [], []
+
+    _add_columns_to_groundtruth_and_prediction_table(
+        df=groundtruth_df, grouper_mappings=grouper_mappings
+    )
+    _add_columns_to_groundtruth_and_prediction_table(
+        df=prediction_df, grouper_mappings=grouper_mappings
+    )
+
+    merged_groundtruths_and_predictions_df = _get_joint_df(
+        prediction_df=prediction_df, groundtruth_df=groundtruth_df
+    )
+
+    cm_counts_df, confusion_matrices = _calculate_confusion_matrix_df(
+        merged_groundtruths_and_predictions_df=merged_groundtruths_and_predictions_df
+    )
+
+    metrics_per_grouper_key_and_grouper_value_df = (
+        _calculate_metrics_at_grouper_value_level(cm_counts_df=cm_counts_df)
+    )
+
+    metrics_to_output += _calculate_precision_recall_f1_metrics(
+        metrics_per_grouper_key_and_grouper_value_df=metrics_per_grouper_key_and_grouper_value_df
+    )
+
+    metrics_to_output += _calculate_accuracy_metrics(cm_counts_df=cm_counts_df)
+
+    metrics_to_output += _calculate_rocauc(
+        prediction_df=prediction_df, groundtruth_df=groundtruth_df
+    )
+
+    # handle type error
+    assert metrics_to_return
```

Comment:
nit: use `raise` instead of `assert`. See [reference](https://www.reddit.com/r/learnpython/comments/zd217y/is_using_assert_considered_good_practice/).